### PR TITLE
Add expected associations to complement missing associations

### PR DESCRIPTION
**Pull request by elainethale**
## Description

Adds expected_associations as a compliment to missing_associations. Dataset contributors can easily describe which combinations of data naturally go together (e.g., sectors and subsectors or sectors and end uses) using expected_associations and then use dsgrid to help them locate and check remaining corner cases (e.g., building types not present in given locations, when new technologies first start showing up) using missing_associations.

Companion test data PR: https://github.com/dsgrid/dsgrid-test-data/pull/53

## Checklist

- [x] Tests exercising the new feature or bug fix
- [x] All tests pass
- [x] At least one code review approval
- [x] Consider transferring TODOs to GitHub

### CODE REVIEW

File: dsgrid/registry/duckdb_data_store.py [hunk 1]
```diff
@@ -181,7 +214,17 @@ def _replace_table(self, df: DataFrame, schema: str, table_name: str) -> None:
...
-    def _list_dim_associations_table_names(self, dataset_id: str, version: str) -> list[str]:
+    def _list_expected_associations_table_names(self, dataset_id: str, version: str) -> list[str]:
+        with self._connect() as con:
+            short_name = _make_table_short_name(dataset_id, version)
+            query = f"""
+                SELECT table_name
+                FROM information_schema.tables
+                WHERE table_schema = '{TABLE_TYPE_TO_SCHEMA["expected_dimension_associations"]}' AND table_name LIKE '%{short_name}%'
+            """
+            return [row[0] for row in con.sql(query).fetchall()]
```
[the remark this answers is not in the document]
Comment (elainethale) [reply dsgrid/registry/duckdb_data_store.py:225]: The SQL query matches what we already have for `_list_missing_associations_table_names`.

File: dsgrid/registry/dataset_registry_manager.py [hunk 2]
```diff
@@ -553,6 +563,33 @@ def _read_lookup_table_from_user_path(
...
+        for path in expected_paths:
+            if path.suffix.lower() == ".parquet":
+                add_df(path)
+            elif path.is_dir():
+                for file_path in path.iterdir():
+                    if file_path.suffix.lower() in (".csv", ".parquet"):
+                        add_df(file_path)
+            elif path.suffix.lower() in (".csv", ".parquet"):
+                add_df(path)
+        return dfs
```
[the remark this answers is not in the document]
Comment (elainethale) [reply dsgrid/registry/dataset_registry_manager.py]: It tries to read any .csv or .parquets and should give error messages for any such files that don't match the expected formats. This behavior is reasonably well documented (i.e., I don't think it's necessary to warn/error on files found with different extensions).

File: docs/source/user_guide/how_tos/index.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ Quick reference guides for specific tasks and operations.
 
 browse_registry
 how_to_dimensions
-how_to_missing_associations
+how_to_dimension_associations
```
[the remark this answers is not in the document]
Comment (elainethale) [reply docs/source/user_guide/how_tos/index.md:10]: I'm not having any issues with the Sphinx toctree. I built the docs from scratch without error, and the page is there and linked to as expected.

**Review by daniel-thom — APPROVED** [review]
